tests/trace/test_op_coroutines.py

```diff
@@ -0,0 +1,220 @@
+import asyncio
+from typing import Coroutine
+
+import pytest
+
+import weave
+from weave.trace.weave_client import Call
+
+
+def test_non_async_non_coro(client):
+    @weave.op()
+    def non_async_non_coro():
+        return 1
+
+    res = non_async_non_coro()
+    assert res == 1
+    res, call = non_async_non_coro.call()
+    assert isinstance(call, Call)
+    assert res == 1
+
+
+def test_non_async_non_coro_method(client):
+    class TestClass:
+        @weave.op()
+        def non_async_non_coro(self):
+            return 1
+
+    test_inst = TestClass()
+    res = test_inst.non_async_non_coro()
+    assert res == 1
+    res, call = test_inst.non_async_non_coro.call(test_inst)
+    assert isinstance(call, Call)
+    assert res == 1
+
+
+@pytest.mark.asyncio
+async def test_non_async_coro(client):
+    @weave.op()
+    def non_async_coro():
+        return asyncio.to_thread(lambda: 1)
+
+    res = non_async_coro()
+    assert isinstance(res, Coroutine)
+    assert await res == 1
+    res, call = non_async_coro.call()
+    assert isinstance(call, Call)
+    assert isinstance(res, Coroutine)
+    assert await res == 1
+
+
+@pytest.mark.asyncio
+async def test_non_async_coro_method(client):
```

Comment:
`test_sync_method_in_coroutine`

Reply:
changed everything to `test_[Op_type]_[return_type]`